tests/test_utilities.py

```diff
@@ -84,6 +86,48 @@ def test_read(tmpdir, use_pathlib):
     assert multi[1].n_blocks == 2
 
 
+@mock.patch('pyvista.utilities.fileio.standard_reader_routine')
+def test_read_legacy(srr_mock):
+    srr_mock.return_value = pyvista.read(ex.planefile)
+    pyvista.read_legacy('legacy.vtk')
+    args, kwargs = srr_mock.call_args
+    reader = args[0]
+    assert isinstance(reader, vtk.vtkDataSetReader)
+    assert reader.GetFileName().endswith('legacy.vtk')
+
+    # check error is raised when no data returned
+    srr_mock.reset_mock()
+    srr_mock.return_value = None
+    with pytest.raises(RuntimeError):
+        pyvista.read_legacy('legacy.vtk')
+
+
+@pytest.mark.parametrize('auto_detect', (True, False))
+@mock.patch('pyvista.utilities.fileio.standard_reader_routine')
```

Comment:
Nice usage of the test mock.

Would it make sense to add in an actual file?  We can add it to the pyvista examples repo and then download it for the tests.

Reply:
Thanks, it worked out well in this situation.

I'm not sure about adding an example file. `read_plot3d()` just delegates to a VTK reader, and it seems that the test strategy to date has been to just trust those. It might open up a can of worms to try to test (and provide examples for) the many different VTK readers out there. Happy to provide an example file if you'd like, but there are many variations even just for Plot3D files - binary/ascii, big/little endian, single/multiblock, 2D/3D, etc.

Reply:
Good point.  Let's call it good at mock then.